app/lib/osm-p2p.js

```diff
@@ -71,13 +59,20 @@ function osmp2p(createOsmDb) {
     }
 
     doc.tags = doc.tags || {};
-    const nodeId = doc.tags["osm-p2p-id"];
+    let nodeId = doc.nodeId;
 
     doc.tags.deviceId = DeviceInfo.getUniqueID();
     doc.timestamp = new Date().toISOString();
     doc.type = "observation";
 
-    observationDb.create(doc, opts, onObservationCreated);
+    if (nodeId) {
+      observationDb.create(doc, opts, onObservationCreated);
+    } else {
+      createNode({ lat: doc.lat, lon: doc.lon }, function(err, node, a, b, c) {
+        doc.nodeId = nodeId = node.value.k;
```

Comment:
I don't think anything requires `doc.nodeId` to be set on the observation. You should be able to change this to just `nodeId = node.value.k`.